core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaMetadata.java

```diff
@@ -305,6 +305,13 @@ private Set<QualifiedTablePrefix> calculatePrefixesWithTableName(
                     })
                     .filter(objectName -> predicate.isEmpty() || predicate.get().test(asFixedValues(objectName)))
                     .map(QualifiedObjectName::asQualifiedTablePrefix)
+                    // In method #getPrefixes, if the prefix set returned by this method has size larger than MAX_PREFIXES_COUNT,
+                    // we will overwrite it with #defaultPrefixes. Limiting the stream at MAX_PREFIXES_COUNT + 1 elements helps
+                    // skip unnecessary computation because we know the resulting set will be discarded when more than MAX_PREFIXES_COUNT
+                    // elements are present. Since there may be duplicate prefixes, a distinct operator is applied to the stream,
+                    // otherwise the stream may be truncated incorrectly.
```

Comment:
This is nice!

should there be some form of comment in the other section changed?

Reply:
@findepi I will add a comment in the final return statement of `#calculatePrefixesWithTableName` that briefly states the same.